bitcoin/src/lib.rs

```diff
@@ -183,35 +183,125 @@ impl LockedTransaction {
     }
 }
 
+fn parse_bitcoin_network(src: &str) -> Result<Network, Error> {
+    match src {
+        "main" => Ok(Network::Bitcoin),
+        "test" => Ok(Network::Testnet),
+        "regtest" => Ok(Network::Regtest),
+        _ => Err(Error::InvalidBitcoinNetwork),
+    }
+}
+
+/// Connect to a bitcoin-core full node or timeout.
+async fn connect(rpc: &Client, connection_timeout: Duration) -> Result<Network, Error> {
```

Comment:
Would it make sense to make this a private function in `BitcoinCoreBuilder`? Just to help with organization

Reply:
It is possible to put functions in impls that do not take a `self` argument. But I don't feel super strongly enough about it, so approving..

Reply:
Yah but it's a private function anyway so I don't think it'll add much to the code organization.